flang/include/flang/Optimizer/Dialect/FIROps.td

```diff
@@ -77,13 +77,11 @@ def fir_BoxCharType : Type<CPred<"$_self.isa<fir::BoxCharType>()">,
 def fir_BoxProcType : Type<CPred<"$_self.isa<fir::BoxProcType>()">,
     "box procedure type">;
 
-def AnyBoxLike : TypeConstraint<Or<[fir_BoxType.predicate,
+def AnyBoxLike : TypeConstraint<Or<[BoxType.predicate,
```

Comment:
Is dropping the prefix really going to work for multiply defined classes like `IntegerType`?

It's better to be overly specific, if possible (it may not be), than to lean on namespace resolution given experience with previous compilation failures.

Reply:
Right. I didn't think about it. Let me update the PR and patches to keep the prefix before merging anything. I'll update what has landed as well.